js/modules/k6/http/http_request.go

```diff
@@ -342,6 +346,43 @@ func (h *HTTP) request(ctx context.Context, rt *goja.Runtime, state *common.Stat
 		},
 	}
 
+	statsSamples := []stats.Sample{}
+	// if digest authentication option is passed, make an initial request to get the authentication params to compute the authorization header
+	if auth == "digest" {
+		tracer := netext.Tracer{}
+		h.debugRequest(state, req, "DigestRequest")
+		res, err := client.Do(req.WithContext(netext.WithTracer(ctx, &tracer)))
```

Comment:
Continuing from my other comment, I think we should keep the whole `if auth == "digest"` section and require users to specify it like they do with `curl`. Aslo, I think we should try to remove the `Authorization` header from the first HTTP request we make to the target when we know the user wants to use `digest` auth.

Reply:
Yes, I agree. If the user has explicitly asked for digest then we shouldn't (or rather let Go's http lib: https://golang.org/src/net/http/client.go#L239) send the `Authorization` header for basic auth, so remove the username/password from the URL.